src/rulesets/asyncapi/schemas/schema.asyncapi2.json

```diff
@@ -0,0 +1,1599 @@
+      "description": "An object to hold a set of reusable objects for different aspects of the AsyncAPI Specification.",
+      "additionalProperties": false,
+      "properties": {
+        "schemas": {
+          "$ref": "#/definitions/schemas"
+        },
+        "messages": {
+          "$ref": "#/definitions/messages"
+        },
+        "securitySchemes": {
+          "type": "object",
+          "patternProperties": {
+            "^[\\w\\d\\.\\-_]+$": {
+              "oneOf": [
+                {
+                  "$ref": "#/definitions/Reference"
+                },
+                {
+                  "$ref": "#/definitions/SecurityScheme"
+                }
+              ]
+            }
+          }
+        },
+        "parameters": {
+          "$ref": "#/definitions/parameters"
+        },
+        "correlationIds": {
+          "type": "object",
+          "patternProperties": {
+            "^[\\w\\d\\.\\-_]+$": {
+              "oneOf": [
+                {
+                  "$ref": "#/definitions/Reference"
+                },
+                {
+                  "$ref": "#/definitions/correlationId"
+                }
+              ]
+            }
+          }
+        },
+        "operationTraits": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/operationTrait"
+          }
+        },
+        "messageTraits": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/messageTrait"
+          }
+        },
+        "serverBindings": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/bindingsObject"
+          }
+        },
+        "channelBindings": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/bindingsObject"
+          }
+        },
+        "operationBindings": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/bindingsObject"
+          }
+        },
+        "messageBindings": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/bindingsObject"
+          }
+        }
+      }
+    },
+    "schemas": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/schema"
+      },
+      "description": "JSON objects describing schemas the API uses."
+    },
+    "messages": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/message"
+      },
+      "description": "JSON objects describing the messages being consumed and produced by the API."
+    },
+    "parameters": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/parameter"
+      },
+      "description": "JSON objects describing re-usable channel parameters."
+    },
+    "schema": {
+      "$id": "#schemaObject",
```

Comment:
@fmvilas and another `$id` here. 

In the spec, the `schemaFormat` is defined as _"A string containing the name of the schema format used to define the message payload. If omitted, implementations should parse the payload as a Schema object. Check out the supported schema formats table for more information. Custom values are allowed but their implementation is OPTIONAL. A custom value MUST NOT refer to one of the schema formats listed in the table."_

The way I interpreted it is that when `schemaFormat` is `undefined` then the payload should be valid against `#/definitions/schema`. 

Similarly, any `example` (or `default`) of that payload should be also valid against both the the `payload` schema and `#/definitions/schema`.

As such, I decorated both the AsyncApi2 schema and the Schema Object with a JsonSchema $id to be able to look them up more easily.

Would have I properly understood the spec, would you consider updating the upstream spec with some `$id` as well? That might make the job of other validators a bit easier.

Reply:
@fmvilas Hmmm. I got that wrong. Only the payload schema should be valid against the SchemaObject. `default` and `example`, `examples` should be valid against the payload schema only. I'll fix that in a following commit.

Reply:
@fmvilas The valid-payload-* rules and tests have been fixed.

Reply:
Awesome! I created an issue so we don't forget: https://github.com/asyncapi/asyncapi/issues/373. Thanks!